dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/filter-fee-record-payment-entity-groups.test.ts

```diff
@@ -1,301 +1,285 @@
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withReference('BBBB').build()],
         };
 
+        const thirdGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('33333333')],
+          payments: [
+            PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+              .withReference('AAAABB') // partially matches firstGroup.payments[0].reference
+              .build(),
+          ],
+        };
+
+        const groups = [firstGroup, secondGroup, thirdGroup];
+
         const paymentReferenceFilter = 'AAAA';
-        const paymentDetailsFilters = { paymentReference: paymentReferenceFilter };
+        const filters = { paymentReference: paymentReferenceFilter };
 
         // Act
-        const result = filterFeeRecordPaymentEntityGroupsByPaymentDetailsFilters(
-          [...allFeeRecordPaymentGroups, anotherFeeRecordPaymentGroup],
-          paymentDetailsFilters,
-        );
+        const result = filterFeeRecordPaymentEntityGroups(groups, filters);
 
         // Assert
-        expect(result).toContainEqual(firstFeeRecordPaymentGroup);
-        expect(result).not.toContainEqual(secondFeeRecordPaymentGroup);
-        expect(result).toContainEqual(anotherFeeRecordPaymentGroup);
+        expect(result).toContainEqual(firstGroup);
+        expect(result).not.toContainEqual(secondGroup);
+        expect(result).toContainEqual(thirdGroup);
       });
     });
 
     it('returns all groups which match all supplied filters', () => {
       // Arrange
-      const anotherPayments = [
-        PaymentEntityMockBuilder.forCurrency('EUR')
-          .withId(15)
-          .withReference('AAAABB') // partially matches firstPayments[0].reference
-          .build(),
-      ];
-      const anotherFeeRecords = [
-        FeeRecordEntityMockBuilder.forReport(aUtilisationReport())
-          .withId(feeRecordIdGenerator.next().value)
-          .withFacilityId('77772222') // partially matches firstFeeRecords[1].facilityId
-          .withPayments(anotherPayments)
-          .build(),
-      ];
-      const anotherFeeRecordPaymentGroup: FeeRecordPaymentEntityGroup = {
-        feeRecords: anotherFeeRecords,
-        payments: anotherPayments,
+      const firstGroup: FeeRecordPaymentEntityGroup = {
+        feeRecords: [aFeeRecordWithFacilityId('11111111')],
+        payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withReference('AAAA').build()],
+      };
+
+      const secondGroup: FeeRecordPaymentEntityGroup = {
+        feeRecords: [aFeeRecordWithFacilityId('22222222')],
+        payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withReference('BBBB').build()],
       };
 
-      const paymentDetailsFilters: PaymentDetailsFilters = {
+      const thirdGroup: FeeRecordPaymentEntityGroup = {
+        feeRecords: [
+          aFeeRecordWithFacilityId('77772222'), // partially matches secondGroup.feeRecords[0].facilityId
```

Comment:
I don't think this comment really adds much and we should avoid unnecessary comments where possible